docs/topics/collections-overview.md

```diff
@@ -6,9 +6,9 @@ of items (possibly zero) that are significant to the problem being solved and ar
 Collections are a common concept for most programming languages, so if you're familiar with, for example, Java or Python 
 collections, you can skip this introduction and proceed to the detailed sections. 
 
-A collection usually contains a number of objects (this number may also be zero) of the same type. Objects in a collection
-are called _elements_ or _items_. For example, all the students in a department form a collection that can be used to
-calculate their average age. 
+A collection usually contains a number of objects (this number may also be zero) of the same type. However if you need them to,
+collections can contain objects of different types. Objects in a collection are called _elements_ or _items_. For example,
```

Comment:
This sentence about objects of different types is making me somewhat uneasy. Is this supposed to be a didactic simplification? I need some background on this, because it feels like it should read something more like "of the same type (and its subtypes)" or similar, but I'm curious if this phrasing aims at a particular goal.

Reply:
This was due to a knowledge gap at the time of writing. (See earlier comment about `Any` type). I've removed it now.